git_machete/tests/functional/test_machete.py

```diff
@@ -26,6 +29,148 @@ def get_head_commit_hash() -> str:
     return os.popen("git rev-parse HEAD").read().strip()
 
 
+class FakeCommandLineOptions(CommandLineOptions):
+    def __init__(self) -> None:
+        super().__init__()
+        self.opt_no_interactive_rebase: bool = True
+        self.opt_yes: bool = True
+
+
+class GitAPIState:
+    pulls: List[Dict[str, Any]] = [{'head': {'ref': 'bugfix/remove'}, 'user': {'login': 'github_user'}, 'base': {'ref': 'develop'}, 'number': '15', 'html_url': 'www.github.com'}]
+    user: Dict[str, str] = {'login': 'github_user', 'type': 'User', 'company': 'VirtusLab'}
+    issues: List[Dict[str, Any]] = []
+    remote_branches: List[str] = []
+
+    def __init__(self, url: str, headers: Dict[str, str] = None, data: Union[str, bytes, None] = None, method: str = ''):
+        self.url: str = url
+        self.status_code: Optional[int] = None
+        self.json_data: Union[str, bytes] = data
+        self.return_data: Optional[Union[List[Dict[str, Any]], Dict[str, Any]]] = None
+        self.headers: Dict[str, str] = headers
+        if method == "GET":
+            self.handle_get()
+        elif method == "PATCH":
+            self.handle_patch()
+        elif method == "POST":
+            self.handle_post()
+
+    def read(self) -> Union[str, bytes]:
+        return json.dumps(self.return_data).encode()
+
+    def handle_get(self) -> None:
+        if 'pulls' in self.url:
+            if '?head' in self.url:
```

Comment:
Instead of this brittle low-level parsing... I'd rather use some library to parse the URL (esp. extract query params from after `?`)